ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -388,11 +401,28 @@ def uplink_switches(self):
 
     @cached_property
     def uplink_interfaces(self):
-        return get(self._switch_data_combined, "uplink_interfaces")
+        return self._range_expand(default(
+            get(self._switch_data_combined, "uplink_interfaces"),
+            get(self.platform_settings, "default_interfaces.uplink"),
+            [])
+        )
 
     @cached_property
     def uplink_switch_interfaces(self):
-        return get(self._switch_data_combined, "uplink_switch_interfaces")
+        auto_uplink_interfaces = []
+        if self.uplink_switches:
+            for uplink_switch in self.uplink_switches:
+                uplink_switch_facts = get(self._hostvars,
+                                          f"avd_switch_facts..{uplink_switch}..switch", required=False,
+                                          org_key=f"avd_switch_facts.({uplink_switch}).switch",
+                                          separator="..")
+                if len(uplink_switch_facts._default_downlink_interfaces) > 0:
+                    auto_uplink_interfaces.append(uplink_switch_facts._default_downlink_interfaces[self.id - 1])
```

Comment:
since you reference this variable from another switch, I think we should make this a regular variable like `switch.downlink_interfaces` and maybe in the future allow setting this manually under the `node`

Reply:
Sure, that makes sense.